### PR TITLE
Small fixes and improvements

### PR DESCRIPTION
**Pull request by proggga**
Small fixes, noting to add in changes

### CODE REVIEW

**Comment by davidism** [comment]
What does this fix? Many of these changes seem unrelated to each other. Many of them are incorrect, or shuffle things around for no apparent reason.

Looking through your other recent PRs, it seems you're making these sorts of changes to a lot of projects. Please be respectful of maintainers time by not making unnecessary changes and clearly explaining the reasons for the changes you do make.

**Comment by proggga** [comment]
> What does this fix? Many of these changes seem unrelated to each other. Many of them are incorrect, or shuffle things around for no apparent reason.
> 
> Looking through your other recent PRs, it seems you're making these sorts of changes to a lot of projects. Please be respectful of maintainers time by not making unnecessary changes and clearly explaining the reasons for the changes you do make.

I really like to improve readablity of code. This is my fetish. All my code is correct. Look carefully. Most of projects like this one. Please recheck:

this one will fix logic and improve speed (bool is faster then work with string)
https://github.com/pallets/flask/pull/3799/files#diff-fed8939d1905b99bba605ad91e9ccdf5ede6223e03ccbc3f0121853035051e62R651

this code incapsulate ad-hoc constant to the one place in code (ye better to create const somewhere, but I don't know the right place)
https://github.com/pallets/flask/pull/3799/files#diff-fed8939d1905b99bba605ad91e9ccdf5ede6223e03ccbc3f0121853035051e62R760

access to right char is faster then slice
https://github.com/pallets/flask/pull/3799/files#diff-1f51c8ded4d4ff7e13badab599ef22436c529c2b5f9c25dc6250c1f9fd985440R295

multiple update faster then many access, will be optimized by python
https://github.com/pallets/flask/pull/3799/files#diff-fed8939d1905b99bba605ad91e9ccdf5ede6223e03ccbc3f0121853035051e62R651

style changes, else and elif after return is not very good
https://github.com/pallets/flask/pull/3799/files#diff-fed8939d1905b99bba605ad91e9ccdf5ede6223e03ccbc3f0121853035051e62R651
https://github.com/pallets/flask/pull/3799/files#diff-fed8939d1905b99bba605ad91e9ccdf5ede6223e03ccbc3f0121853035051e62R651
https://github.com/pallets/flask/pull/3799/files#diff-fed8939d1905b99bba605ad91e9ccdf5ede6223e03ccbc3f0121853035051e62R651

here I removed also 'else'
https://github.com/pallets/flask/pull/3799/files#diff-fed8939d1905b99bba605ad91e9ccdf5ede6223e03ccbc3f0121853035051e62R651

**Comment by proggga** [comment]
> Many of them are incorrect
If I've made incorrect change, please say where, I don't want to break something. If I've made this, only by mistake.

I really try help with style, small mistakes, improve readability and code support (remove unneeded tabs when you have return + else/elif

may be can fix this PR in some case

File: src/flask/blueprints.py [hunk 1]
```diff
@@ -284,11 +284,11 @@ def merge_dict_nested(self_dict, app_dict):
 
         if cli_resolved_group is None:
             app.cli.commands.update(self.cli.commands)
-        elif cli_resolved_group is _sentinel:
-            self.cli.name = self.name
-            app.cli.add_command(self.cli)
         else:
-            self.cli.name = cli_resolved_group
+            if cli_resolved_group is _sentinel:
```
Comment (ThiefMaster) [inline src/flask/blueprints.py:288]: This change does not improve readability. Repeating a line once (or even twice) is perfectly fine; being overly DRY actually makes code harder to read here.

File: src/flask/cli.py [hunk 2]
```diff
@@ -709,7 +706,7 @@ def convert(self, value, param, ctx):
         except click.BadParameter:
             value = click.STRING(value, param, ctx).lower()
 
-            if value == "adhoc":
+            if _isAdHoc(value):
```
Comment (ThiefMaster) [inline src/flask/cli.py:709]: camelCase has no place in Python code (see PEP8). Also, you replace a very simple check with a function call where you now need to check the function first to see what it does!

File: src/flask/cli.py [hunk 3]
```diff
@@ -760,6 +757,11 @@ def _validate_key(ctx, param, value):
     return value
 
 
+def _isAdHoc(cert):
+    """Check is it adhoc"""
```
Comment (ThiefMaster) [inline src/flask/cli.py:761]: This docstring is the type of documentation that nobody really needs :) It's like "loop over list" above a loop...

File: src/flask/helpers.py [hunk 4]
```diff
@@ -292,7 +292,7 @@ def external_url_handler(error, endpoint, values):
         url_adapter = reqctx.url_adapter
         blueprint_name = request.blueprint
 
-        if endpoint[:1] == ".":
+        if endpoint[0] == ".":
```
Comment (ThiefMaster) [inline src/flask/helpers.py:295]: You are changing behavior here:

```python
>>> empty = ''
>>> empty[:1]
''
>>> empty[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: string index out of range
```

File: src/flask/helpers.py [hunk 5]
```diff
@@ -340,10 +340,14 @@ def external_url_handler(error, endpoint, values):
...
-        values["_method"] = method
-        values["_scheme"] = scheme
+        values.update(
+            {
+                "_external": external,
+                "_anchor": anchor,
+                "_method": method,
+                "_scheme": scheme,
+            }
+        )
```
Comment (ThiefMaster) [inline src/flask/helpers.py:350]: This is longer and doesn't look better. The usecase for `dict.update()` is when you already have a sequence...

File: src/flask/helpers.py [hunk 6]
```diff
@@ -841,38 +845,37 @@ def _find_package_path(root_mod_name):
         if spec.origin in {"namespace", None}:
             return os.path.dirname(next(iter(spec.submodule_search_locations)))
         # a package (with __init__.py)
-        elif spec.submodule_search_locations:
+        if spec.submodule_search_locations:
             return os.path.dirname(os.path.dirname(spec.origin))
         # just a normal module
-        else:
-            return os.path.dirname(spec.origin)
+        return os.path.dirname(spec.origin)
```
Comment (ThiefMaster) [inline src/flask/helpers.py:851]: With the old code every return was on the same levl - now the last one is on a different level. Which option to choose is completely personal preference though, but I don't think sending PRs to projects you are not related to do make changes that are purely based on personal preference is helpful.

**Review by ThiefMaster — COMMENTED** [review]
I'm someone who cares a lot about clean and readable code. Your PR does not really help in that regard - I added some comments explaining why that's the case.

File: src/flask/helpers.py (hunk 4 above)
Comment (proggga) [reply src/flask/helpers.py:295]: Hm, thats true, made a mistake

File: src/flask/cli.py (hunk 3 above)
Comment (proggga) [reply src/flask/cli.py:761]: Yea, Its my problem from go, start write everywhere camecase, can fx this

**Comment by proggga** [comment]
Please tell, what about other changes.
I can revert some changes

**Comment by ThiefMaster** [comment]
I don't think it makes much sense to do anything with this PR - none of the changes are really worth a PR...

**Comment by proggga** [comment]
> I don't think it makes much sense to do anything with this PR - none of the changes are really worth a PR...

Thanks, Ive really tried.

**Comment by davidism** [comment]
If you're interested in contributing, there are plenty of open issues with stuff we're actually asking for. However, we won't accept random changes like this.

**Pull request closed without merging**